### ISSUE TITLE
ExifTool Commands: Command fails. Quote Problem?

### ISSUE THREAD

**Issue by jjuser** [hvdwolf/jExifToolGUI#95]
I try to rename QuickTime movie files (*.MOV) by creation date. But I didn't manage it. In the MacOS terminal the following ist working:
`exiftool -d %Y%m%d_%H%M%%-c_%%f.%%e "-filename<QuickTime:CreateDate" 'DSCN9534.MOV'
`
I get: 20180325_1441_DSCN9534.MOV

In jExifToolGUI > ExifTool Commands > Parameters i tried:

`-d %Y%m%d_%H%M%%-c_%%f.%%e "-filename<QuickTime:CreateDate"`

Result:
```
============= "/Users/jejatimax/Desktop/test mov metadata/test2/Neuer Ordner/DSCN9534.MOV" =============
Warning: No writable tags set from /Users/jejatimax/Desktop/test mov metadata/test2/Neuer Ordner/DSCN9534.MOV
Warning: Invalid tag name 'quicktime:createdate"' - /Users/jejatimax/Desktop/test mov metadata/test2/Neuer Ordner/DSCN9534.MOV
    0 image files updated
    1 image files unchanged
```
Next try with single-quotes:
` '-d %Y%m%d_%H%M%%-c_%%f.%%e "-filename<QuickTime:CreateDate"'`

Result:
```
============= "/Users/jejatimax/Desktop/test mov metadata/test2/Neuer Ordner/DSCN9534.MOV" =============
Error: File not found - '-d %Y%m%d_%H%M%%-c_%%f.%%e "-filename<QuickTime:CreateDate"'
======== /Users/jejatimax/Desktop/test mov metadata/test2/Neuer Ordner/DSCN9534.MOV
ExifTool Version Number         : 12.10
File Name                       : DSCN9534.MOV
Directory                       : /Users/jejatimax/Desktop/test mov metadata/test2/Neuer Ordner
File Size                       : 1338 kB
File Modification Date/Time     : 2020:11:23 21:52:58+01:00
File Access Date/Time           : 2020:11:23 21:53:26+01:00
File Inode Change Date/Time     : 2020:11:23 23:01:37+01:00
File Permissions                : rwxrwxrwx
File Type                       : MOV
File Type Extension             : mov
MIME Type                       : video/quicktime
Major Brand                     : Apple QuickTime (.MOV/QT)
Minor Version                   : 2007.9.0
Compatible Brands               : qt  , niko
Movie Header Version            : 0
Modify Date                     : 2018:03:25 14:41:34
Time Scale                      : 30000
Duration                        : 1.17 s
Preferred Rate                  : 1
Preferred Volume                : 100.00%
Preview Time                    : 0 s
Preview Duration                : 0 s
Poster Time                     : 0 s
Selection Time                  : 0 s
Selection Duration              : 0 s
Current Time                    : 0 s
Next Track ID                   : 3
Track Header Version            : 0
Track Create Date               : 2018:03:25 14:41:34
Track Modify Date               : 2018:03:25 14:41:34
Track ID                        : 1
Track Duration                  : 1.17 s
Track Layer                     : 0
Track Volume                    : 0.00%
Image Width                     : 1280
Image Height                    : 720
Graphics Mode                   : ditherCopy
Op Color                        : 32768 32768 32768
Compressor ID                   : avc1
Source Image Width              : 1280
Source Image Height             : 720
X Resolution                    : 72
Y Resolution                    : 72
Compressor Name                 : AVC Coding
Bit Depth                       : 24
Color Representation            : nclc 6 1 6
Video Frame Rate                : 29.97
Matrix Structure                : 1 0 0 0 1 0 0 0 1
Media Header Version            : 0
Media Create Date               : 2018:03:25 14:41:34
Media Modify Date               : 2018:03:25 14:41:34
Media Time Scale                : 48000
Media Duration                  : 1.15 s
Balance                         : 0
Handler Class                   : Data Handler
Handler Type                    : Alias Data
Audio Format                    : mp4a
Purchase File Format            : mp4a
Maker Note Version              : Nikon 0.1.0.0
Make                            : NIKON
Model                           : COOLPIX P7800
Software                        : COOLPIX P7800  V1.0
Create Date                     : 2018:03:25 14:41:34
Date/Time Original              : 2018:03:25 14:41:34
Frame Count                     : 35
Frame Rate                      : 29.97
Frame Width                     : 1280
Frame Height                    : 720
Audio Channels                  : 2
Audio Bits Per Sample           : 16
Audio Sample Rate               : 48000
GPS Version ID                  : 2.3.0.0
GPS Latitude Ref                : Unknown ()
GPS Longitude Ref               : Unknown ()
GPS Altitude Ref                : Above Sea Level
GPS Time Stamp                  : 00:00:00
GPS Img Direction Ref           : Unknown ()
GPS Img Direction               : undef
GPS Map Datum                   : 
GPS Date Stamp                  : 0000:00:00
Thumbnail Image                 : (Binary data 5430 bytes, use -b option to extract)
Preview Image                   : (Binary data 235411 bytes, use -b option to extract)
XMP Toolkit                     : Image::ExifTool 12.10
Label                           : 2018:03:25 14:41:34
Media Data Size                 : 1123546
Media Data Offset               : 246525
Image Size                      : 1280x720
Megapixels                      : 0.922
GPS Date/Time                   : 0000:00:00 00:00:00Z
GPS Latitude                    : 
GPS Longitude                   : 
Avg Bitrate                     : 7.7 Mbps
Rotation                        : 0
    1 image files read
    1 files could not be read
```

**Comment by hvdwolf** [comment]
To put this correctly, you need to use:
`-d %Y%m%d_%H%M%%-c_%%f.%%e '-filename<QuickTime:CreateDate'`

Around the redirect character, either < or >, you need single quotes on MacOS and Linux. This is because the shell interpreter on MacOS/Linux works that way.

However: you are right. There is some bug in this screen due to this < or > character. I will check that.

On the other hand: My workflow would be:
- load file(s)
- select file(s)
- Correct file date (I copy my images/movies from my camera via WiFi to my laptop. Therefore the file date is no longer correct)
![setfiledatetodatetimeoriginal](https://user-images.githubusercontent.com/4520361/100064802-4291e700-2e33-11eb-8a5e-a3f602eeb439.jpg)
- Rename files: (menu) Renaming -> Rename photos. In your case:
![rename](https://user-images.githubusercontent.com/4520361/100065920-aff24780-2e34-11eb-9726-2bb18b47fa50.jpg)

I just did that with some MP4 files to test whether that still works, and it works for me,

**Comment by jjuser** [comment]
Hi, thank you for your answer!
Unfortunately your first proposal does not work:
`-d %Y%m%d_%H%M%%-c_%%f.%%e '-filename<QuickTime:CreateDate'`
i get:
```
============= "/Users/jejatimax/Desktop/test mov metadata/test 2020-11-25/DSCN9534.MOV" =============
Warning: No writable tags set from /Users/jejatimax/Desktop/test mov metadata/test 2020-11-25/DSCN9534.MOV
Warning: Invalid tag name 'quicktime:createdate'' - /Users/jejatimax/Desktop/test mov metadata/test 2020-11-25/DSCN9534.MOV
    0 image files updated
    1 image files unchanged
```

Your other solution is partially working:
- adjusting date is working
- renaming isn't working, when there are spaces in the path
 
![Bildschirmfoto 2020-11-25 um 23 31 23](https://user-images.githubusercontent.com/42519963/100288493-17b0ab80-2f77-11eb-9136-c51811b822fd.png)

When removing the spaces from the test path, renaming ist working. But the result has a double suffix like "20180325_144134_DSCN9534.MOV.MOV"

As most of my image and video files are stored in directories with spaces in its name, this is unfortunately not yet a solution for me.

The files of interest are located in various subfolders. To rename them in one go I searched for the files in the mac os finder and dragged the resulting list to jExifToolGUI V1.7.1. Now I want to correct and rename them.

![Bildschirmfoto 2020-11-25 um 23 53 15](https://user-images.githubusercontent.com/42519963/100289621-9eff1e80-2f79-11eb-8afc-03bc07e47cf4.png)

**Comment by hvdwolf** [comment]
Hi,

> Unfortunately your first proposal does not work:
> -d %Y%m%d_%H%M%%-c_%%f.%%e '-filename<QuickTime:CreateDate'
That's also what I mentioned in the next line. There is an issue with the redirect sign.

This "spaces" issue was solved in one of the earliest releases. I don't know why or how it came back, but I will solve it asap.

**Comment by hvdwolf** [comment]
"Renaming with spaces" issue: solved (again).
Will be in next release.

**Comment by jjuser** [comment]
Thanks!

**Comment by klewis-us57** [comment]
I think there is a problem that is raised by the behavior above, and I am seeing it as well.  It seems that any parameter that is not a Tag is not recognized in the Command string.  The **-d** parameter is called out as an invalid Tag.  If I use ExifTool on the command line, it works just fine.  

if I use this command in the ExifTool Command tab: (with testname)
**-testname<Movie-${MediaCreateDate;}-%f.%e'**
I get back: 
_============= "/Users/kenlewis/Pictures/2011/12/IMG_0899.MOV" =============
'/Users/kenlewis/Pictures/2011/12/IMG_0899.MOV' --> '/Users/kenlewis/Pictures/2011/12/Movie-20111218 112306-IMG_0899.MOV''
    0 image files updated
    1 image files unchanged_

The file does get changed, without any formatting changes. 

if I add the format string with -d 
**-d %Y-%m-%d '-testname<${MediaCreateDate;}-%f.%e' **

And I get : 

============= "/Users/kenlewis/Pictures/2011/12/IMG_0899.MOV" =============
Warning: No writable tags set from /Users/kenlewis/Pictures/2011/12/IMG_0899.MOV
Warning: **Invalid tag name 'd %y-%m-%d '-testname'** - /Users/kenlewis/Pictures/2011/12/IMG_0899.MOV
    0 image files updated
    1 image files unchanged

**Comment by hvdwolf** [comment]
Hi,
Thanks for your report. I did the same tests and so did jjuser.
The issue has been introduced a few versions ago when I added extra formatting options when users request "tabbed", "html" of "htmldump" output as part of their exiftool command, which means that I now "parse" the command differently.
I think I have a solution now, without "removing" these new output options, but I need to fully test first.
I already wanted to release a new version adding new functionality, last weekend, but due to this bug that was postponed. Hopefully this weekend.
